### ISSUE TITLE
Sheets not positioned properly if the window has a horizontal scroll bar 

### ISSUE THREAD

**Issue by lowPolySkeleton** [DestinyItemManager/DIM#10261]
### DIM Version

current app and beta 

### Browser Details

tested in chrome and FF

### OS Details

win 10 

### Describe the bug

![image](https://github.com/DestinyItemManager/DIM/assets/29002828/07fe4a56-7500-4ef1-b7e5-88aa9919af12)
when opening a sheet, if there is a horizontal scroll bar the sheet leaves a gap between the bottom the page and the sheet as seen above. Scaling the window till there is no scroll bar makes this gap disappear, scaling it back to having a scroll bar make it re-appear.

**Comment by liamdebeasi** [comment]
It looks like DIM is setting the [`bottom` value of the sheet](https://github.com/DestinyItemManager/DIM/blob/bf67d40a0c94866574b85340bda2c1d88f5dd6f1/src/app/dim-ui/Sheet.m.scss#L18) equal to that of `--viewport-bottom-offset`. This value changes [when the visual viewport resizes](https://github.com/DestinyItemManager/DIM/blob/bf67d40a0c94866574b85340bda2c1d88f5dd6f1/src/app/css-variables.ts#L67) which can happen when the horizontal scrollbar appears. As a result, the sheet is shifted upwards by an amount equal to the height of the horizontal scrollbar.

I did a quick search, and it looks like DIM only uses `--viewport-bottom-offset` [for the sheet](https://github.com/search?q=repo%3ADestinyItemManager%2FDIM%20setCssVariableEventListeners&type=code). According to the comment around where the CSS Variable is set, this is done so the on-screen keyboard does not overlap the sheet.

------

One option for fixing this would be to only set the `--viewport-bottom-offset` value if it's over a certain threshold. I implemented a [similar fix for Ionic Framework](https://github.com/ionic-team/ionic-framework/blob/34d4055c6b4361c9c6ac9ee6b5a0a539c4a3d575/core/src/utils/keyboard/keyboard.ts#L87-L95).

Is this an approach the project maintainers would be comfortable with? I'd be happy to make a PR.
